web/src/NetworkListItem.jsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React from "react";
+
+import {
+  Card,
+  CardBody,
+  Radio,
+  Spinner,
+  Split,
+  SplitItem,
+  Text
+} from "@patternfly/react-core";
+
+import {
+  EOS_LOCK as LockIcon,
+  EOS_SIGNAL_CELLULAR_ALT as SignalIcon
+} from "eos-icons-react";
+
+import { ConnectionState } from "./client/network/model";
+
+import Center from "./Center";
+import WifiNetworkMenu from "./WifiNetworkMenu";
+import WifiConnectionForm from "./WifiConnectionForm";
+
+const networkState = (state) => {
+  switch (state) {
+    case ConnectionState.ACTIVATING:
+      return 'Connecting';
+    case ConnectionState.ACTIVATED:
+      return 'Connected';
+    case ConnectionState.DEACTIVATING:
+      return 'Disconnecting';
+    case ConnectionState.DEACTIVATED:
+      return 'Disconnected';
+    default:
+      return "";
+  }
+};
+
+function NetworkListItem ({ network, isChecked, isFocused, showAsChecked, showSpinner, onSelect, onCancel }) {
```

Comment:
It would be nice to document the arguments.

Reply:
Ok, let's do it